sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -247,3 +247,76 @@ case class DecimalAddNoOverflowCheck(
       newLeft: Expression, newRight: Expression): DecimalAddNoOverflowCheck =
     copy(left = newLeft, right = newRight)
 }
+
+/**
+ * A divide expression for decimal values which is only used internally by Avg.
+ *
+ * It will fail when nullOnOverflow is false follows:
+ *   - left is null, as the right side should never be null
```

Comment:
since we only invoke `DecimalDivideWithOverflowCheck` when count > 0, I think left (`sum` in AVG) will never be null?

Reply:
ah, yes. We skip the count == 0 and the merging expression do not check the overflow of sum. That said here the sum will never be null.

Reply:
@cloud-fan in fact, it will produce null value if sum overflow. We do not check the overflow during merging, but the max precison of Spark is limited by 38.. which means the sum value may over 1E38BD